### DISCUSSION TITLE
I need to understand more about signing transactions.

### DISCUSSION THREAD

**Discussion by AtegyekaRodgers** [reach-sh/reach-lang#1001] [Q&A] [answered]
I need to understand more about signing transactions. 
Some time back, I had an issue such that whenever I would try to deploy the contract, the execution would halt and get stack at the point where a promise had to be resolved. I later found out that it was waiting for me to respond to a prompt on my pera-algo app, which required that operation to be "signed" before taking effect. 

Now after successfully deploying the contract and getting the correct contract info, I again see a similar behavior (halting) when a participant attaches to the contract and tries to make a call to any API function. The execution halts at the exact line of code that is supposed to make the call and receive feedback from the backed. But this time unlike previously, my pera wallet app does not show any prompt for me to "sign" the transaction.

I suspect may be there is still some kind of "signing" I have to do which it keeps waiting for indefinitely. The promise never gets resolved, I forever wait for the response from the API function.

You can checkout my code on github, specifically look at lines 28 - 36 inside src/dashboard.js 
https://github.com/Reach-Insurance/react/blob/main/src/dashbord.js
![apis-halt](https://user-images.githubusercontent.com/47475700/168895729-dd5aab5a-86c5-4173-8136-980dba75672d.png)

**Comment by jeapostrophe** [comment, marked as answer]
Any time you tell Reach to make a publication or API call, then the user has to authenticate that they want this to happen. This is a signing. Reach will communicate to your wallet, which is in charge of taking control of the application interface and telling the user that this needs to happen. Reach cannot control this process, because it has no access to the wallet's display at all. This is a security feature of Web3 progams.
